simple_graph.py

```diff
@@ -27,9 +28,20 @@ def add_edge(self, node1, node2):
             self.graph.setdefault(node1, [node2])
 
     def del_node(self, node):
-        self.graph.pop(node, None)
+        """Delete a node from the graph. Throw a KeyError if the node wasn't
+        in the graph."""
+        try:
+            self.graph.pop(node)
+        except KeyError:
+            raise KeyError("{} does not exist".format(node))
+        for k, v in self.graph.iteritems():
+            if node in v:
+                v.remove(node)
 
     def del_edge(self, node1, node2):
+        """Delete an edge from the graph. Throw a KeyError if the first node
+         wasn't in the graph.  Throw a ValueError if the second node wasn't
+         in the graph."""
```

Comment:
Properly formatted docstring would have the first sentence by itself on the first line, and then would put the rest onto later lines.  (see [PEP257](https://www.python.org/dev/peps/pep-0257/))